pkg/offline/offline.go

```diff
@@ -157,6 +167,164 @@ func WithQueue(filepath string, syncLimit int) (heartbeat.HandleOption, error) {
 	}, nil
 }
 
+// Sync returns a function to send queued heartbeats to the WakaTime API.
+func Sync(filepath string, syncLimit int) func(next heartbeat.Handle) error {
+	return func(next heartbeat.Handle) error {
+		log.Debugf("execute offline sync with file %s", filepath)
+
+		hh, err := popHeartbeats(filepath, sendLimit)
+		if err != nil {
+			return fmt.Errorf("failed to fetch heartbeat from offline queue: %s", err)
+		}
+
+		if len(hh) == 0 {
+			log.Debugln("abort execution, as there are no queued heartbeats ready for sending")
+
+			return nil
+		}
+
+		results, err := next(hh)
+		if err != nil {
+			var count int
+
+			for {
+				if count >= maxRequeueAttempts {
+					log.Errorf("abort requeuing after %d unsuccessful attempts", count)
```

Comment:
Let's change this to something like:

```go
log.Errorf("failed writing heartbeat to offline queue: %s", string(json.Marshal(hh)[0]))
```

That code probably doesn't work, but shows what we want is to log the heartbeat to `~/.wakatime.log` when unable to save to the offline db. That allows parsing the log file to recover lost heartbeats.